.gitlab-ci.yml

```diff
@@ -0,0 +1,74 @@
+stages:
```

Comment:
@compiaffe just doing a quick first pass here (and thank you for your contribution).

I think `.gitlab-ci.yml` should probably live outside of this repo, since it is specific to building and publishing images into a private registry.

Reply:
Certainly, if possible I'd like to see/setup the github alternative so that you folks at balena build the containers.

Reply:
FYI: It pushes to our public registry at https://gitlab.com/aivero/open-source/open-balena/container_registry